File: TradeItIosEmsApi/TradeItBrokerAccount.m
#import "TradeItBrokerAccount.h"

@implementation TradeItBrokerAccount

- (id)initWithAccountBaseCurrency:(NSString *)accountBaseCurrency
                    accountNumber:(NSString *)accountNumber
                             name:(NSString *)name
                         tradable:(BOOL)tradable {
    if (self = [super init]) {
        self.accountBaseCurrency = accountBaseCurrency;
        self.accountNumber = accountNumber;
        self.name = name;
        self.tradable = tradable;
    }

    return self;
}

- (NSString *)description {
    return [
        NSString stringWithFormat:@"accountBaseCurrency=%@ accountNumber=%@ accountName=%@ tradable=%d",
        self.accountBaseCurrency,
        self.accountNumber,
        self.name,
        self.tradable
    ];
}

@end

File: TradeItIosTicketSDK2/TradeItBrokerAccount.swift
class TradeItBrokerAccount: Codable {
    var accountBaseCurrency: String
    var accountNumber: String
    var accountIndex: String
    var name: String
    var tradable: Bool
    var userCanDisableMargin: Bool
    var orderCapabilities: [TradeItInstrumentOrderCapabilities]
    
    init(accountBaseCurrency: String, accountNumber: String, name: String, tradable: Bool) {
        self.accountBaseCurrency = accountBaseCurrency
        self.accountNumber = accountNumber
        self.name = name
        self.tradable = tradable
        self.userCanDisableMargin = false
        self.orderCapabilities = []
        self.accountIndex = ""
    }
}
